### ISSUE TITLE
FileSystem transport - file locking

### ISSUE THREAD

**Issue by mgibas** [rebus-org/Rebus#306]
I'm using filesystem transport and have a problem problem with locking files:

```csharp
The process cannot access the file 'XXXXXXXX.rebusmessage' because it is being used by another process.
```

It's situation where two processes want to send message to same queue. For that we have this piece of code:

FileSystemMessageQueue.cs
```csharp
public void Send(string destinationQueueName, TransportMessageToSend message, ITransactionContext context)
{
    EnsureQueueInitialized(destinationQueueName);

    var destinationDirectory = GetDirectoryForQueueNamed(destinationQueueName);

    var serializedMessage = Serialize(message);
    var fileName = GetNextFileName();
    var fullPath = Path.Combine(destinationDirectory, fileName);

    Action commitAction = () => File.WriteAllText(fullPath, serializedMessage, FavoriteEncoding);

    if (context.IsTransactional)
    {
        context.DoCommit += commitAction;
    }
    else
    {
        commitAction();
    }
}
```

in this line we are interlocking msg number, but thats not enough if we talk about multiple processes
```csharp
var fileName = GetNextFileName();
```

Any ideas how to overcome this ? Maybe some king of name mangling (with i.e. guid ?)

**Comment by mgibas** [comment]
After quick "live" test it seams that switching from:

```csharp
string GetNextFileName()
{
    return string.Format("{0:yyyy_MM_dd_HH_mm_ss}_{1:0000000}.rebusmessage",
        DateTime.UtcNow, Interlocked.Increment(ref incrementingCounter));
}
```
to
```csharp
string GetNextFileName()
{
    return string.Format("{0}_{1:0000000}.rebusmessage",
        Guid.NewGuid(), Interlocked.Increment(ref incrementingCounter));
}
```
works as expected. Let me know if this change makes any sens - I will create PR :)

**Comment by asgerhallas** [comment]
Yes, I believe the easiest way is to concat some uniqueness at the end of the filename. That will make the order of concurrent messages random, but that - hopefully - shouldn't matter to the application. It could be suffixed with a guid or maybe the bus instance name.

**Comment by asgerhallas** [comment]
Oh, I was too slow there... I believe you should let the guid be suffixed to keep the order. I'm not sure, but I guess the messages are read in order by name.

**Comment by mgibas** [comment]
Ya, good point:
```csharp
var fileNames = Directory.GetFiles(GetDirectoryForQueueNamed(inputQueue), "*.rebusmessage")
                    .OrderBy(f => f)
                    .ToList();
```

**Comment by mookid8000** [comment]
should be solved by #307 :heart_eyes:
